lib/cdo/shared_constants.rb

```diff
@@ -0,0 +1,29 @@
+# This file is generated via generateSharedConstants.rb. DO NOT CHANGE MANUALLY
+
+module SharedConstants
+  # Used to communicate different types of levels
+  LevelKind = OpenStruct.new(
+    {
+      "peer_review": "peer_review",
+      "assessment": "assessment",
+      "puzzle": "puzzle",
+      "unplugged": "unplugged",
+      "level": "level"
+    }
+  )
+
+  # Different possibilities for level.status, used to communicate how user has performed on a given level
+  LevelStatus = OpenStruct.new(
+    {
+      "not_tried": "not_tried",
+      "submitted": "submitted",
+      "locked": "locked",
+      "perfect": "perfect",
+      "passed": "passed",
+      "attempted": "attempted",
+      "review_accepted": "review_accepted",
+      "review_rejected": "review_rejected",
+      "dots_disabled": "dots_disabled"
+    }
+  )
```

Comment:
Yeah I can't find a RuboCop rule for hash keys being symbols, but it's a stated preference even on their [styleguide](https://github.com/github/rubocop-github/blob/master/STYLEGUIDE.md#collections). It's not that important since this is just input to OpenStruct, but it would be terser and more standard to write it as symbols.

`level_status.unplugged` is more standard. Or to define it as a constant, `LEVEL_STATUS`.